cmd/init.go

```diff
@@ -153,14 +153,17 @@ var initCmd = &cobra.Command{
 
 func init() {
 	rootCmd.AddCommand(initCmd)
-	initCmd.PersistentFlags().StringVar(&etcdAdmConfig.Name, "name", "", "etcd member name")
-	initCmd.PersistentFlags().StringVar(&etcdAdmConfig.Version, "version", constants.DefaultVersion, "etcd version")
-	initCmd.PersistentFlags().StringVar(&etcdAdmConfig.ReleaseURL, "release-url", constants.DefaultReleaseURL, "URL used to download etcd")
-	initCmd.PersistentFlags().StringVar(&etcdAdmConfig.CertificatesDir, "certs-dir", constants.DefaultCertificateDir, "certificates directory")
-	initCmd.PersistentFlags().StringSliceVar(&etcdAdmConfig.ServerCertSANs, "server-cert-extra-sans", etcdAdmConfig.ServerCertSANs, "optional extra Subject Alternative Names for the etcd server signing cert, can be multiple comma separated DNS names or IPs")
-	initCmd.PersistentFlags().StringVar(&etcdAdmConfig.InstallDir, "install-dir", constants.DefaultInstallDir, "install directory")
-	initCmd.PersistentFlags().StringVar(&etcdAdmConfig.Snapshot, "snapshot", "", "Etcd v3 snapshot file used to initialize member")
-	initCmd.PersistentFlags().BoolVar(&etcdAdmConfig.SkipHashCheck, "skip-hash-check", false, "Ignore snapshot integrity hash value (required if copied from data directory)")
-	initCmd.PersistentFlags().DurationVar(&etcdAdmConfig.DownloadConnectTimeout, "download-connect-timeout", constants.DefaultDownloadConnectTimeout, "Maximum time in seconds that you allow the connection to the server to take.")
-	initCmd.PersistentFlags().StringArrayVar(&etcdAdmConfig.EtcdDiskPriorities, "disk-priorities", constants.DefaultEtcdDiskPriorities, "Setting etcd disk priority")
+	flags := initCmd.PersistentFlags()
+	flags.StringVar(&etcdAdmConfig.Name, "name", "", "etcd member name")
+	flags.StringVar(&etcdAdmConfig.Version, "version", constants.DefaultVersion, "etcd version")
+	flags.StringVar(&etcdAdmConfig.ReleaseURL, "release-url", constants.DefaultReleaseURL, "URL used to download etcd")
+	flags.StringVar(&etcdAdmConfig.CertificatesDir, "certs-dir", constants.DefaultCertificateDir, "certificates directory")
+	flags.StringSliceVar(&etcdAdmConfig.ServerCertSANs, "server-cert-extra-sans", etcdAdmConfig.ServerCertSANs, "optional extra Subject Alternative Names for the etcd server signing cert, can be multiple comma separated DNS names or IPs")
```

Comment:
Does this flag already do what you need for server certs?

Reply:
Damn, I'm sorry -- I got bitten by "too much pasting, not enough reading" since I had that patch against the latest released version of etcdadm (v0.1.2) and I didn't _read_ before pasting those lines

Given that both `init.go` and `join.go` now correctly contain `ServerCertSANs`, plus the perfectly valid "why expose PeerSANs" question below, then I think this PR can be closed.

---

That said, it looks like the ServerCertSANs just missed the v0.1.2 release window, so is a fresh release of `etcdadm` coming? And if so, could I beg and plead to have that next release to have the usual go binaries attached to it, in order to keep one from having to build `etcdadm` locally?

Reply:
Thanks for explaining. I will make an effort this week to include go binaries in the release.